concurrency/src/tasks/time.rs

```diff
@@ -22,9 +22,16 @@ where
 {
     let cancellation_token = CancellationToken::new();
     let cloned_token = cancellation_token.clone();
+    let gen_server_cancellation_token = handle.cancellation_token();
     let join_handle = rt::spawn(async move {
-        let _ = select(
+        // Timer action is ignored if it was either cancelled or the associated GenServer is no longer running.
+        let cancel_conditions = select(
```

Comment:
Ah, nice! I like having both, the individual cancellation_token, and the gen_server associated one